syntax = "proto3";
import "common.proto";
import "backup.proto";
package huawei.proto.control;

service BackupControl {
    rpc CreateBackup(CreateBackupReq) returns(CreateBackupAck){}
    rpc DeleteBackup(DeleteBackupReq) returns(DeleteBackupAck){}
    rpc RestoreBackup(RestoreBackupReq) returns(RestoreBackupAck){}
    rpc QueryBackup(QueryBackupReq) returns(QueryBackupAck){}
    rpc ListBackup(ListBackupReq) returns(ListBackupAck){}
}

message CreateBackupReq {
    string vol_name = 1; 
    uint64 vol_size = 2;
    string backup_name = 3;
    BackupOption backup_option = 4;
}

message CreateBackupAck {
    StatusCode status = 1;
}

message DeleteBackupReq {
    string vol_name = 1; 
    string backup_name = 2;
}

message DeleteBackupAck {
    StatusCode status = 1;
}

message RestoreBackupReq {
    string vol_name = 1; 
    string backup_name = 2;
    string new_vol_name = 3;
    uint64 new_vol_size = 4;
    string new_block_device =5;
}

message RestoreBackupAck {
    StatusCode status = 1;
}

message QueryBackupReq {
    string vol_name = 1; 
    string backup_name = 2;
}

message QueryBackupAck {
    StatusCode status = 1;
    BackupStatus backup_status = 2;
}

message ListBackupReq {
    string vol_name = 1; 
}

message ListBackupAck {
    StatusCode status = 1;
    repeated string backup_name = 2;
}
